### PR TITLE
WRO-8315: Update dependencies to React 18 and the latest enact

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
1.Update enact templates to React 18 
2. Added hydrateRoot where applicable
3. Updated enact dependencies
4. Fixed some lint warnings

**NOTE**: electron template didn't start even with the former setup. There was an error "global is not defined". Fixed the error by setting       _webPreferences: {        nodeIntegration: true,          contextIsolation: false,        }_
The desktop app started succesfuly afterwards 
image

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-8315

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian [daniel.stoian@lgepartner.com](mailto:daniel.stoian@lgepartner.com)

### CODE REVIEW

File: packages/electron/template/main/index.js [hunk 1]
```diff
@@ -18,15 +18,19 @@ function createWindow () {
 		height: 600,
 		icon: (process.platform==='win32'
 				? path.join(__dirname, 'assets', 'icon-mini.png')
-				: path.join(__dirname, 'assets', 'icon.png'))
+				: path.join(__dirname, 'assets', 'icon.png')),
+		webPreferences: {
```
Comment (daniel-stoian-lgp) [inline packages/electron/template/main/index.js:22]: this was modified in order to make the destop app start

**Review by alexandrumorariu — APPROVED** [review]
LGTM

File: packages/electron/template/main/index.js [hunk 2]
```diff
@@ -18,15 +18,19 @@ function createWindow () {
...
+			nodeIntegration: true,
+			contextIsolation: false,
+		}
```
Comment (daniel-stoian-lgp) [inline packages/electron/template/main/index.js:25]: this was added in order to make the app start successfuly

**Review by adrian-cocoara-lgp — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@enact/templates",
-  "version": "4.5.0-rc.1",
+  "version": "4.5.0",
```
Comment (MikyungKim) [inline package.json]: We change this version when releasing. :)
I'll revert these.

File: packages/electron/template/main/index.js (hunk 2 above)
Comment (MikyungKim) [reply packages/electron/template/main/index.js:25]: 🤔 I'm still seeing the "global is not defined". Let's take a look at this in another ticket.

**Review by MikyungKim — APPROVED** [review]
:lgtm:

**Pull request merged**